src/app.js

```diff
@@ -1 +1,19 @@
-// write code here
+const fs = require('fs');
+const path = require('path');
+
+(() => {
+  try {
+    const [source, dest] = process.argv.slice(2);
+
+    let finalPath = dest;
+
+    if (fs.existsSync(dest) && fs.statSync(dest).isDirectory()) {
+      finalPath = path.resolve(dest, path.basename(source));
+    }
+
+    fs.renameSync(source, finalPath);
```

Comment:
Using `fs.renameSync` can throw an error if the source file does not exist or if there are permission issues. You might want to add checks to ensure the source file exists before attempting to rename it, or handle specific error types in the catch block.